azurelinuxagent/ga/exthandlers.py

```diff
@@ -1388,6 +1361,8 @@ def set_handler_status(self, status="NotReady", message="", code=0):
         try:
             handler_status_json = json.dumps(get_properties(handler_status))
             if handler_status_json is not None:
+                if not os.path.exists(state_dir):
```

Comment:
Currently if there is any exceptions in extension processing and the config directory doesnt exist, the error is never reported back to CRP and the customer has to wait for 90 mins before the operation timesout. This is a bad Customer Experience. To mitigate it, I'm creating the config directory incase it doesnt exist while reporting status.